jmetal-core/src/main/java/org/uma/jmetal/operator/impl/crossover/IntegerSBXCrossover.java

```diff
@@ -165,6 +103,84 @@ public void setCrossoverProbability(double crossoverProbability) {
     return offspring;
   }
 
+private void biggerthanEPS(IntegerSolution parent1, List<IntegerSolution> offspring, int i, int valueX1, int valueX2) {
+	double rand;
+	double y1;
+	double y2;
+	double yL;
+	double yu;
+	double c1;
+	double c2;
+	double alpha;
+	double beta;
+	double betaq;
+	if (Math.abs(valueX1 - valueX2) > EPS) {
+
+	    if (valueX1 < valueX2) {
+	      y1 = valueX1;
+	      y2 = valueX2;
+	    } else {
+	      y1 = valueX2;
+	      y2 = valueX1;
+	    }
+
+	    yL = parent1.getLowerBound(i);
+	    yu = parent1.getUpperBound(i);
+	    rand = randomGenerator.getRandomValue();
+	    beta = 1.0 + (2.0 * (y1 - yL) / (y2 - y1));
+	    alpha = 2.0 - Math.pow(beta, -(distributionIndex + 1.0));
+
+	    betaq = randless(rand, alpha);
+
+	    c1 = 0.5 * ((y1 + y2) - betaq * (y2 - y1));
+	    beta = 1.0 + (2.0 * (yu - y2) / (y2 - y1));
+	    alpha = 2.0 - Math.pow(beta, -(distributionIndex + 1.0));
+
+	    betaq = randless(rand, alpha);
+
+	    c2 = 0.5 * (y1 + y2 + betaq * (y2 - y1));
+
+	    c1 = checkC(yL, yu, c1);
+	    
+	    c2 = checkC(yL, yu, c2);
+
+	    randomless(offspring, i, c1, c2);
+	  } else {
+	    offspring.get(0).setVariableValue(i, valueX1);
+	    offspring.get(1).setVariableValue(i, valueX2);
+	  }
+}
+
+private double checkC(double yL, double yu, double c) {
+	if (c < yL) {
+	  c = yL;
+	}else if (c > yu) {
+	  c = yu;
+	}
+	return c;
```

Comment:
This method is just horrible!
First, we can get rid of the useless assertions:
```java
if (c < yL) {
  return yL;
}else if (c > yu) {
  return yu;
}
```
Second, this code enforce a value to be within an interval, which is a trivial min-max operation that can be done in 1 line:
```java
return Math.min(Math.max(yL, c), yu);
```

We may like or not the use of the single-line version, but at least the method itself can be clearly refactored to be way clearer, something like:
```java
private double forceInterval(double min, double max, double value) {...}
```
and because the same operation is done on `c1` and `c2`, it actually makes sense to create a closure on min & max, and to apply it on the values, for example:
```java
private Function<Double, Double> intervalLimiter(double min, double max) {
		return (value) -> Math.min(Math.max(min, value), max);
}
```
then at the beginning of the loop:
```java
Function<Double, Double> limiter = intervalLimiter(
  parent1.getLowerBound(i),
  parent1.getUpperBound(i)
);
```
and where needed:
```java
c1 = limiter.apply(c1)
c2 = limiter.apply(c2)
```

Such a refactoring would seem to me way more explicit, easy to understand, and modern.